toxcore/group_chats.c

```diff
@@ -8270,14 +8274,21 @@ static void group_delete(GC_Session *c, GC_Chat *chat)
         c->chats_index = i;
 
         if (!realloc_groupchats(c, c->chats_index)) {
-            LOGGER_ERROR(chat->log, "Failed to reallocate groupchats array");
+            LOGGER_ERROR(c->messenger->log, "Failed to reallocate groupchats array");
         }
     }
 }
 
 int gc_group_exit(GC_Session *c, GC_Chat *chat, const uint8_t *message, uint16_t length)
 {
-    const int ret =  group_can_handle_packets(chat) ? send_gc_self_exit(chat, message, length) : 0;
+    chat->flag_exit = true;
+    return group_can_handle_packets(chat) ? send_gc_self_exit(chat, message, length) : 0;
+}
+
+non_null()
+static int kill_group(GC_Session *c, GC_Chat *chat)
+{
+    const int ret = gc_group_exit(c, chat, nullptr, 0);
     group_delete(c, chat);
```

Comment:
You now call `group_delete()` twice. The `gc_group_exit()` call a line above in here sets `chat->flag_exit` for a delayed `group_delete()` in ` if (chat->flag_exit) {` of `do_gc()`, and you also call `group_delete()` in here right away. Is that intended or is this `group_delete()` in here is redundant?

Reply:
`kill_group()` is only called after a `tox_kill()`. `do_gc()` never runs again after that so the flag will never be seen. But even if that weren't the case, the group would already be deleted and the flag reset by the next interation.